src/legacy/core_plugins/data/public/actions/select_range_action.ts

```diff
@@ -0,0 +1,91 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { i18n } from '@kbn/i18n';
+import {
+  IAction,
+  createAction,
+  IncompatibleActionError,
+} from '../../../../../plugins/ui_actions/public';
+// @ts-ignore
+import { onBrushEvent } from './filters/brush_event';
+import {
+  esFilters,
+  FilterManager,
+  TimefilterContract,
+  changeTimeFilter,
+  extractTimeFilter,
+  mapAndFlattenFilters,
+} from '../../../../../plugins/data/public';
+// eslint-disable-next-line @kbn/eslint/no-restricted-paths
+import { getIndexPatterns } from '../../../../../plugins/data/public/services';
+
+export const SELECT_RANGE_ACTION = 'SELECT_RANGE_ACTION';
+
+interface ActionContext {
+  data: any;
```

Comment:
Can this be further typed out?  The types are so critical to the trigger and what actions can be attached to it.

Reply:
I'm going to create an issue to track this